### ISSUE TITLE
Be robust against strange input from the helper

### ISSUE THREAD

**Issue by hans-ekbrand** [hans-ekbrand/lc0#9]
In the below example `bestmove` is `info` which must a an error, but the PV seems to get extracted correctly anmyway, which is reasonable since the PV is extracted from the line before the `bestmove` line
```
0122 09:16:10.357260 139938523772480 ../../src/mcts/auxengine.cc:589] thread: 1 auxe:info depth 12 seldepth 26 multipv 1 score cp 112 time 70 nodes 45855 nps 655071 tbhits 0 hashfull 11 pv g7c7 
0122 09:16:10.370359 139931970168384 ../../src/mcts/auxengine.cc:649] pv:info depth 21 seldepth 37 multipv 1 score cp 62 time 1169 nodes 1156841 nps 989598 tbhits 7 hashfull 69 pv g6g7 f6e4 d2d3 b7c6 a4a1 e4f6 d5f6 d8f6 g7c7 c6g2 a1g1 g2h3 c4b5 e8f8 b5c4 f8e8 c7f7 e5e4 d3e4 f6b2 c4d5 a8c8 f7a7 b2e5 e3g5 h3d7 a7a5 h8h2 a5a7 
0122 09:16:10.370391 139931970168384 ../../src/mcts/auxengine.cc:650] bestanswer:info
0122 09:16:10.370425 139931970168384 ../../src/mcts/auxengine.cc:433] debug info: length of PV given to helper engine: 0 position given to helper: position fen r2bk2r/1b4p1/3p1nR1/p2Np3/R1B1P3/4B2P/1PPK4/8 w kq - 1 23 white to move at root, length of my_moves_from_the_white_side 20 my_moves_from_the_white_side: g6g7 f3e5 d2d3 b2c3 a4a1 e5f3 d5f6 d1f3 g7c7 c3g7 a1g1 g7h6 c4b5 e1f1 b5c4 f1e1 c7f7 e4e5 d3e4 f3b7 
```
Thus far it works fine, but after that there are even more bizarre input:
```
0122 09:16:10.370481 139931970168384 ../../src/mcts/auxengine.cc:649] pv:
0122 09:16:10.370486 139931970168384 ../../src/mcts/auxengine.cc:650] bestanswer:
0122 09:16:10.370491 139931970168384 ../../src/mcts/auxengine.cc:653] Empty PV, returning early from doAuxEngine().
0122 09:16:10.370496 139931970168384 ../../src/mcts/auxengine.cc:457] DoAuxEngine() called for node Term:0 This:0x7f3e4c1a28a0 Parent:0x7f3e4c004bb0 Index:0 Child:0x7f31f8218c80 Sibling:0 WL:-0.241216 N:5807 N_:1748 Edges:46 Bounds:-1,1 So...
```
Empty PV returning early when root node is explored leads to a bad (never-ending) loop. In this case we perhaps needs to reset/flush the input and output streams and issue `uci` or `isready`.

Looking for the root cause I see normal output from thread 0

```
0122 09:16:10.219284 139931970168384 ../../src/mcts/auxengine.cc:589] thread: 0 auxe:info depth 21 seldepth 37 multipv 1 score cp 62 time 1169 nodes 1156841 nps 989598 tbhits 7 hashfull 69 pv g6g7 f6e4 d2d3 b7c6 a4a1 e4f6 d5f6 d8f6 g7c7 c6g2 a1g1 g2h3 c4b5 e8f8 b5c4 f8e8 c7f7 e5e4 d3e4 f6b2 c4d5 a8c8 f7a7 b2e5 e3g5 h3d7 a7a5 h8h2 a5a7 
0122 09:16:10.219350 139931970168384 ../../src/mcts/auxengine.cc:433] debug info: length of PV given to helper engine: 0 position given to helper: position fen r2bk2r/1b4p1/3p1nR1/p2Np3/R1B1P3/4B2P/1PPK4/8 w kq - 1 23 white to move at root, length of my_moves_from_the_white_side 20 my_moves_from_the_white_side: g6g7 f3e5 d2d3 b2c3 a4a1 e5f3 d5f6 d1f3 g7c7 c3g7 a1g1 g7h6 c4b5 e1f1 b5c4 f1e1 c7f7 e4e5 d3e4 f3b7 
```
followed by what appears to be a new call to `DoAuxEngine()` by thread 0
```
0122 09:16:10.370434 139931970168384 ../../src/mcts/auxengine.cc:457] DoAuxEngine() called for node Term:0 This:0x7f3e4c1a28a0 Parent:0x7f3e4c004bb0 Index:0 Child:0x7f31f8218c80 Sibling:0 WL:-0.241216 N:5807 N_:1748 Edges:46 Bounds:-1,1 Solid:0
0122 09:16:10.370451 139931970168384 ../../src/mcts/auxengine.cc:545] add pv= from root position: r2bk2r/1b4p1/3p1nR1/p2Np3/R1B1P3/4B2P/1PPK4/8 w kq - 1 23
0122 09:16:10.370465 139931970168384 ../../src/mcts/auxengine.cc:562] Starting infinite query for thread 0
0122 09:16:10.370470 139931970168384 ../../src/mcts/auxengine.cc:563] opstream for thread 0 at: 0x7f4160001520
0122 09:16:10.370476 139931970168384 ../../src/mcts/auxengine.cc:574] Setting auxengine_stopped_ to false for thread 0
0122 09:16:10.370481 139931970168384 ../../src/mcts/auxengine.cc:649] pv:
0122 09:16:10.370486 139931970168384 ../../src/mcts/auxengine.cc:650] bestanswer:
0122 09:16:10.370491 139931970168384 ../../src/mcts/auxengine.cc:653] Empty PV, returning early from doAuxEngine().
```
when the input `bestmove` is received then parsing the loop is exited, but in this case the engine was not stopped which lead to havoc.

To sum up, this flow was observed:
1. works normally; 2 gets the bizarre `bestanswer:info` still succeeds to extract a valid PV; 3. starts over since it had recieved `bestmove`; 3 fails to get anything sensible out of the input stream, since it the engine hasn't stopped.

Solution: whenever bestanswer is `info`, issue `stop`

**Comment by hans-ekbrand** [comment]
Solved by keeping strict control of the output streams.
